### ISSUE TITLE
Failed to connect to Dataverse in ASP .NET Core Web API

### ISSUE THREAD

**Issue by lucaUU** [microsoft/PowerPlatform-DataverseServiceClient#320]
Hi
Im creating an ASP .NET Core Web Api Project and try to connect to dataverse. 
I always get
Failed to connect to Dataverse:
MsalClientException: Only loopback redirect uri is supported, but urn:ietf:wg:oauth:2.0:oob was found. Configure http://localhost or http://localhost:port both during app registration and when you create the PublicClientApplication object. See https://aka.ms/msal-net-os-browser for details

Client code on the controller we connect to crm is
[HttpGet(Name = "GetWeatherForecast")]
        public IEnumerable<WeatherForecast> Get()
        {
            var service = new ServiceClient($@"AuthType=ClientSecret;
                        SkipDiscovery=true;url={_config["crmServiceUrl"]};
                        Secret={_config["crmClientSecretId"]};
                        ClientId={_config["crmClientId"]};
                        RequireNewInstance=true");

            return Enumerable.Range(1, 5).Select(index => new WeatherForecast
            {
                Date = DateTime.Now.AddDays(index),
                TemperatureC = Random.Shared.Next(-20, 55),
                Summary = Summaries[Random.Shared.Next(Summaries.Length)]
            })
            .ToArray();
        }

What am I missing here? If i use the new serviceclient in a console it works with the same code and clientid/secrets ect. Thnaks

**Comment by BetimBeja** [comment]
@lucaUU check your client secret... One colleague of mine had the same error, and in her case the Client Secret was null. Maybe it needs to be XML Escaped in your case?

**Comment by lucaUU** [comment]
@BetimBeja You are correct. ClientSecret was null. Was miss leaded by the error message. Thanks.
